SamplesDashboard/SamplesDashboard/ClientApp/src/components/DetailList.tsx

```diff
@@ -0,0 +1,186 @@
+            columns: columns,
+            items: this._allItems,
+            selectionDetails: this._getSelectionDetails(),
+            announcedMessage: undefined
+        };
+    }
+
+    componentDidMount = () => {
+        this.fetchData();
+    }
+
+    //fetching the data from the api
+    fetchData = async () => {
+        const response = await fetch('api/Samples');
+        const data = await response.json();
+        this._allItems = data;
+        this.setState(
+            {
+                items: data,
+                selectionDetails: this._getSelectionDetails()
+            });
+    }
+
+    public render(): JSX.Element {
+        const { columns, items, selectionDetails, announcedMessage } = this.state;
+
+    return (
+        <Fabric>
+            <div className={classNames.wrapper}>
+            <div className={detailListClass}>{selectionDetails}</div>
+            <ScrollablePane scrollbarVisibility={ScrollbarVisibility.auto}>
+                <Sticky stickyPosition={StickyPositionType.Header}>
+                <TextField
+                  className={detailListClass}
+                        label="Filter by name:"
+                        onChange={this._onFilterName}
+                  styles={{ root: { maxWidth: '300px' } }}
+                    />
+                    </Sticky>
+                    {announcedMessage ? <Announced message={announcedMessage} /> : undefined}
+                <MarqueeSelection selection={this._selection} data-is-scrollable={true}>
+                <DetailsList
+                    items={items}
+                    columns={columns}
+                    selectionMode={SelectionMode.multiple}
+                    getKey={this._getKey}
+                    setKey="multiple"
+                    layoutMode={DetailsListLayoutMode.justified}
+                    isHeaderVisible={true}
+                    selection={this._selection}
+                    selectionPreservedOnEmptyClick={true}
+                    ariaLabelForSelectionColumn="Toggle selection"
+                    ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+                    checkButtonAriaLabel="Row checkbox"
+                    onItemInvoked={this._onItemInvoked}
+                  />
+                </MarqueeSelection>
+                </ScrollablePane>
+                </div>
+        </Fabric>
+        );
+    }
+
+    private _getKey(item: any, index?: number): string {
+        return item.key;
+    }
+
+  private _getSelectionDetails(): string {
+    const selectionCount = this._selection.getSelectedCount();
+
+    switch (selectionCount) {
+      case 0:
+        return 'No items selected';
+      case 1:
+        return '1 item selected: ' + (this._selection.getSelection()[0] as IListDataItem).name;
+      default:
+        return `${selectionCount} items selected`;
+    }
+  }
+
+    private _onFilterName = (ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, text: string): void => {
```

Comment:
You can fix build issue with
```suggestion
        private _onFilterName = (ev: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, text?: string | undefined): void => {
```

Reply:
Oh okay thank you for this